evm/rlp/blocks.py

```diff
@@ -16,35 +21,35 @@ def get_sedes(cls):
     transaction_class = None
 
     @classmethod
-    def get_transaction_class(cls):
+    def get_transaction_class(cls) -> BaseTransaction:
         if cls.transaction_class is None:
             raise AttributeError("Block subclasses must declare a transaction_class")
         return cls.transaction_class
 
     @classmethod
-    def from_header(cls, header, chaindb):
+    def from_header(cls, header: BlockHeader, chaindb: BaseChainDB) -> 'BaseBlock':
         """
         Returns the block denoted by the given block header.
         """
         raise NotImplementedError("Must be implemented by subclasses")
 
     @property
-    def hash(self):
+    def hash(self) -> bytes:
```

Comment:
@gsalgado what do you think about us creating a custom type for 32 byte hashes (as outlined here: http://mypy.readthedocs.io/en/latest/kinds_of_types.html#newtypes)

This would make things a bit stricter, ensuring that we are in-fact dealing with hashes rather than just any arbitrary `bytes` value.

Reply:
I'd be very much in favor of this!

Reply:
Sounds good to me, but since this would likely be useful in other libraries as well (e.g. eth-utils and eth-keys?), I wonder if maybe it should be defined in one of those instead?

Reply:
Yes, especially because these types will move between layers. For instance the evm may return an `address` to trinity and if these are different NewTypes (one defined in evm, one in trinity) then mypy should complain about it. So moving those into a shared lib makes much sense.

Reply:
In that case, we should probably create a very thin `eth-typing` library with this stuff in it because I don't *think* `eth-utils` isn't a global dependency across all of the `eth` codebases.

Reply:
options:

1. do it locally now and migrate to something from `eth-typing` later.
2. keep it `bytes` now and migrate to something from `eth-typing` later.

I'm fine with either.

Reply:
I'd vote for 2.)